src/pages/config/ConfigDetailsPage.tsx

```diff
@@ -1,40 +1,39 @@
+import { useQuery } from "@tanstack/react-query";
 import { useEffect, useState, useMemo, useCallback } from "react";
-import { useParams, useSearchParams, useOutletContext } from "react-router-dom";
-import { ConfigItem, getConfig } from "../../api/services/configs";
+import { useParams } from "react-router-dom";
+import { getConfig } from "../../api/services/configs";
 import { EvidenceType } from "../../api/services/evidence";
 import { AttachEvidenceDialog } from "../../components/AttachEvidenceDialog";
-import { Button } from "../../components/Button";
+import { ConfigsDetailsBreadcrumbNav } from "../../components/BreadcrumbNav/ConfigsDetailsBreadCrumb";
+import { ConfigDetailsSelectedLinesControls } from "../../components/ConfigsPage/ConfigDetailsSelectedLinesControls";
 import { JSONViewer } from "../../components/JSONViewer";
+import { ConfigLayout } from "../../components/Layout";
 import { Loading } from "../../components/Loading";
 import { toastError } from "../../components/Toast/toast";
+import { usePartialUpdateSearchParams } from "../../hooks/usePartialUpdateSearchParams";
 
 export function ConfigDetailsPage() {
   const { id } = useParams();
-  const [params, setParams] = useSearchParams();
-  const [isLoading, setIsLoading] = useState(true);
-  const [showIncidentModal] = useState(false);
+  const [params, setParams] = usePartialUpdateSearchParams();
   const [attachAsAsset, setAttachAsAsset] = useState(false);
-  const [dialogKey, setDialogKey] = useState(0);
-  const [checked, setChecked] = useState({});
-  const [configDetails, setConfigDetails] = useState<ConfigItem | undefined>();
-  // @ts-ignore
-  const { setTabRight } = useOutletContext();
-
-  useEffect(() => {
-    getConfig(id!)
-      .then((res) => {
-        const data = res?.data?.[0];
-        setConfigDetails(data);
-      })
-      .catch((err) => toastError(err))
-      .finally(() => {
-        setIsLoading(false);
-      });
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [id]);
+  const [checked, setChecked] = useState<Record<string, any>>({});
+
+  const { isLoading, data: configDetails } = useQuery(
```

Comment:
Please move this logic to react queries file where we have all the queries.

Reply:
The one disadvantage of doing this is we lose type inference and autocompletion on options, which is very in cases like [select](https://tkdodo.eu/blog/react-query-data-transformations), which is useful when you want to transform the data without affecting the cached data. I am still trying to figure out a way of wrapping around useQuery, without losing it.

Reply:
Okay, but this will lead to code duplication. we can solve type inference problem later.